test/test_datasets.py

```diff
@@ -2206,6 +2206,58 @@ def inject_fake_data(self, tmpdir: str, config):
         return len(sampled_classes * n_samples_per_class)
 
 
+class FGVCAircraftTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.FGVCAircraft
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        split=("train", "val", "trainval", "test"), annotation_level=("variant", "family", "manufacturer")
+    )
+
+    def inject_fake_data(self, tmpdir: str, config):
+        split = config["split"]
+        annotation_level = config["annotation_level"]
+        annotation_level_to_file = {
+            "variant": "variants.txt",
+            "family": "families.txt",
+            "manufacturer": "manufacturers.txt",
+        }
+
+        root_folder = pathlib.Path(tmpdir) / "fgvc-aircraft-2013b"
+        data_folder = root_folder / "data"
+
+        num_images_per_class = 5
+        classes = ["707-320", "Hawk T1", "Tornado"]
+        num_samples_per_class = 4 if split == "trainval" else 2
+
+        datasets_utils.create_image_folder(
+            data_folder,
+            "images",
+            file_name_fn=lambda idx: f"{idx}.jpg",
+            num_examples=num_images_per_class * len(classes),
+        )
+
+        images_classes = []
+        for i in range(len(classes)):
+            images_classes.extend(
+                [
+                    f"{idx} {classes[i]}"
+                    for idx in random.sample(
+                        range(i * num_images_per_class, (i + 1) * num_images_per_class), num_samples_per_class
+                    )
+                ]
+            )
+
+        annotation_file = root_folder / "data" / annotation_level_to_file[annotation_level]
+        images_annotation_file = root_folder / "data" / f"images_{annotation_level}_{split}.txt"
+
+        with open(annotation_file, "w") as file:
+            file.write("\n".join(classes))
```

Comment:
Can we move this further to the top given that this is somewhat static and does not depend on the other data generated in the loop above?

Reply:
Actually, it has to happen after the `create_image_folder ` part because otherwise it would complain `root_folder / "data"` is not an existing path.